In [1]:
import requests
from bs4 import BeautifulSoup

import numpy as np
import pandas as pd

%matplotlib inline
import matplotlib

In [2]:
url = "http://books.toscrape.com/"

In [3]:
response = requests.get(url, headers={"Accept-Language":"en-US"})
soup = BeautifulSoup(response.content, "html.parser")

< article class="product_pod">

In [6]:
books_html  = soup.find_all("article", class_="product_pod")

## Parsing _one_ book

In [15]:
type(books_html)
len(books_html)

20

In [43]:
#Title
books_html[0].find("h3").find("a").attrs
book_title = books_html[0].find("h3").find("a").attrs["title"]
float(books_html[0].find("p", class_="price_color").string[1:])

51.77

In [53]:
books_html[0].find("p", class_="star-rating")['class'][1]

'Three'

In [54]:
book_stars_html = books_html[0].find("p", class_="star-rating")
book_stars_html

<p class="star-rating Three">
<i class="icon-star"></i>
<i class="icon-star"></i>
<i class="icon-star"></i>
<i class="icon-star"></i>
<i class="icon-star"></i>
</p>

In [44]:
books_html[0].find("p", class_="star-rating")

<p class="star-rating Three">
<i class="icon-star"></i>
<i class="icon-star"></i>
<i class="icon-star"></i>
<i class="icon-star"></i>
<i class="icon-star"></i>
</p>

In [56]:
def parse_rating(rating_classes):
    
    if rating_classes == 'One':
        return 1
    
    elif rating_classes == 'Two':
        return 2
    
    elif rating_classes == 'Three':
        return 3 
    
    elif rating_classes == 'Four':
        return 4
    
    elif rating_classes == 'Five':
        return 5   

    return 0

In [61]:
parse_rating(books_html[3].find("p", class_="star-rating")['class'][1])

4

Title => ["A light in the attic", "Tipping the Velvet", ...]

Price => [51.77, 53.74, ...]

Rating => [3, 1, ...]

In [74]:
books_dict={'Title':[], 'Price':[], 'Rating':[]}


for book in books_html:
    title = book.find("h3").find("a").attrs["title"]
    price = float(book.find("p", class_="price_color").string[1:])
    rating = parse_rating(book.find("p", class_="star-rating")['class'][1])
    
    books_dict['Title'].append(title) 
    books_dict['Price'].append(price)
    books_dict['Rating'].append(rating)

#book_title = books_html[0].find("h3").find("a").attrs["title"]

In [75]:
books_dict

{'Title': ['A Light in the Attic',
  'Tipping the Velvet',
  'Soumission',
  'Sharp Objects',
  'Sapiens: A Brief History of Humankind',
  'The Requiem Red',
  'The Dirty Little Secrets of Getting Your Dream Job',
  'The Coming Woman: A Novel Based on the Life of the Infamous Feminist, Victoria Woodhull',
  'The Boys in the Boat: Nine Americans and Their Epic Quest for Gold at the 1936 Berlin Olympics',
  'The Black Maria',
  'Starving Hearts (Triangular Trade Trilogy, #1)',
  "Shakespeare's Sonnets",
  'Set Me Free',
  "Scott Pilgrim's Precious Little Life (Scott Pilgrim #1)",
  'Rip it Up and Start Again',
  'Our Band Could Be Your Life: Scenes from the American Indie Underground, 1981-1991',
  'Olio',
  'Mesaerion: The Best Science Fiction Stories 1800-1849',
  'Libertarianism for Beginners',
  "It's Only the Himalayas"],
 'Price': [51.77,
  53.74,
  50.1,
  47.82,
  54.23,
  22.65,
  33.34,
  17.93,
  22.6,
  52.15,
  13.99,
  20.66,
  17.46,
  52.29,
  35.02,
  57.25,
  23.88,
  3